docs/threat-model.md

```diff
@@ -0,0 +1,158 @@
+private user/system/operational data that they should not have access to. Further, they may be able
+to construct spoofed requests for abuse in conjunction with Threat (3).
+
+## Defense Mechanisms
+
+### (1, 2) User Authentication
+
+To protect against user impersonation, the application will utilize mechanisms of user authentication
+to validate the user's identity prior to allowing access to any privileged operations or data (including
+operations or data tied to user identity).
+
+#### Password Authentication
+
+Upon user creation, users will select a password to use for authentication. We will impose several
+requirements to ensure the password is not trivially bruteforceable.
+
+1. Minimum of 12 characters; maximum of 60 characters
+2. Includes at least one number
+3. Includes at least one uppercase and lowercase alphabetic character
+4. Includes at least one non-alphabetic character (symbol)
+
+These passwords will be stored in a User Authentication service. They will be salted and hashed to
+protect against offline attacks if the password database were ever leaked.
+
+* The salt will be a cryptographically random string.
+* The chosen hash function will be computationally expensive to deter offline attacks.
+
+TODO: pick hash function. `bcrypt` seems preferred nowadays? it may also handle salting?
+
+#### Two Factor Authentication (2FA)
+
+As user passwords may be shared between multiple services, may be bruteforceable, may be leaked 
+through a variety of means, etc., our application will utilize 2FA to further validate user
+identities.
+
+TODO: mechanism? email?
+
+### (1, 2) User Sessions
+
+It is not practical to require users to authenticate for every request. This presents a poor UX
+and is expensive for our backend services. To alleviate this pressure, we will implement user 
+sessions to permit users to authenticate once and continue submitting requests for a time without
+requiring authentication again.
+
+User sessions will be implemented by a `SessionID` that is a 256 bit-string, the length chosen
+to harden `SessionID`s against bruteforce attacks. These will be randomly generated with
+cryptographically strong randomness. `SessionID`s will be stored in a Redis memory store alongside
+a `UserID` and a session timeout. Sessions will timeout after 12 hours of inactivity.
+
+Users will receive their 256-bit `SessionID` after successful user authentication. This will have
+to be sent with every privileged request in a manner specified by the application API. The API
+Gateway would check the Redis memory store to validate the `SessionID` before serving the request.
+Failure to validate the `SessionID` would result in a failure which would be communicated to the user.
+
+TODO: details of `SessionID` storage (do we want to use Redis?), details of timeout
+
+### (3) Internal Docker Network
+
+Backend services will operate within an internal Docker network that would not have exposed ports
```

Comment:
Note for future: The API gateway will be containerized, so it will access the APIs via the internal Docker network. Docker containers may be able to talk to each other at will, however